specs/network_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,132 @@
+= Specification for adding network details in node hardware inventory
+
+Add network details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently network details are missing in node hardware inventory.
+we have to add network details like IPv4, netmask, status as a part of
+node hardware inventory.
+
+== Use Cases
+
+Tendrl should show the current status of active network interface.
+
+== Proposed change
+* Netifaces is used to find the active network interface details.
+* Add netifaces as dependency.
+* From node_agent add a new function in pull_hardware_inventory to find active
+  network interfaces details except lo, virbr0, virbr0-nic and update those
+  details into node_inventory list.
+* Create a new class in persistence module to keep network details.
+* Add a new function in persister to persist network details.
+* Modify the manager to populate the instance of newly created class using
+  node_inventory list and persist the instance in etcd.
+* Create an object in tendrl_definitions_node_agent for network details.
+
+== Alternatives
+
+* We can also use ethtool for getting network details. But ethool is not
+  available for pip.
+
+== Data model impact
+
+* Add network details in etcd with key nodes/node_id/Network/interface_name.
+* Add network object in tendrl_definitions_node_agent/Network.
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect active network
+  interface details.
+* Create a new class in persistence module to keep the network details.
+* Add a new function in persister to persist network details.
+* Modify the manager modules to read and store active network interface details.
+* Create an object in tendrl_definitions_node_agent for network details.
+* Commands used to retrieve network details are:
+```
+    interfaces = netifaces.interfaces()
+    ipv4_addr_list = netifaces.ifaddresses(interfaces[integer_value])[AF_INET]
+    netmask = ipv4_addr_detail[integer_value]['netmask']
+```
+* Excpected output format for the active network details is:
+```
+{"interface_name": {"ipv4": ["ipv4address", ...],
+                    "netmask": ["subnet", ...],
+                    "status":"up/down"
+                   }, ...
+}
+```
+
+
+=== Tendrl/gluster_integration impact
+
+None
+
+=== Tendrl/ceph_integration impact
+
+None
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
+
+We might have to monitor the status of the network interface(up/down).
+
+== Other end user impact
+
+None
+
+== Performance Impact
+
+None
+
+== Other deployer impact
+
```

Comment:
@GowthamShanmugam , Can you update this section with packaging related changes required. Pm me if you need any help here